gr-trellis/python/trellis/bindings/interleaver_python.cc

```diff
@@ -14,56 +15,81 @@
-             py::arg("K"),
-             py::arg("seed"),
-             D(interleaver, interleaver, 4))
-
-
-        .def("K", &interleaver::K, D(interleaver, K))
-
-
-        .def("INTER", &interleaver::INTER, D(interleaver, INTER))
-
-
-        .def("DEINTER", &interleaver::DEINTER, D(interleaver, DEINTER))
-
-
-        .def("write_interleaver_txt",
-             &interleaver::write_interleaver_txt,
-             py::arg("filename"),
-             D(interleaver, write_interleaver_txt))
-
-        ;
+    auto version =
+        std::make_tuple(std::stoi(gr::major_version()), std::stoi(gr::api_version()));
+    const char* interleaver_arg = "INTER";
+    const char* length_arg = "K";
+    const char* copy_arg = "INTERLEAVER";
+    if (version >= std::make_tuple(3, 11)) {
+        interleaver_arg = "inter";
+        length_arg = "k";
+        copy_arg = "interleaver";
+    }
+
+    auto interleaver_bind =
+        py::class_<interleaver, std::shared_ptr<interleaver>>(
+            m, "interleaver", D(interleaver))
+            .def(py::init<>(), D(interleaver, interleaver, 0))
+            .def(py::init<gr::trellis::interleaver const&>(),
+                 py::arg(copy_arg),
+                 D(interleaver, interleaver, 1))
+            .def(py::init<std::vector<int, std::allocator<int>> const&>(),
+                 py::arg(interleaver_arg),
+                 D(interleaver, interleaver, 2))
+            .def(py::init<char const*>(), py::arg("name"), D(interleaver, interleaver, 3))
+            .def(py::init<unsigned int, int>(),
+                 py::arg(length_arg),
+                 py::arg("seed"),
+                 D(interleaver, interleaver, 4))
+            .def("k", &interleaver::k, D(interleaver, k))
+            .def("interleaver_indices",
+                 &interleaver::interleaver_indices,
+                 D(interleaver, interleaver_indices))
+            .def("deinterleaver_indices",
+                 &interleaver::deinterleaver_indices,
+                 D(interleaver, interleaver_indices))
+            .def("write_interleaver_txt",
+                 &interleaver::write_interleaver_txt,
+                 py::arg("filename"),
+                 D(interleaver, write_interleaver_txt));
+
+    if (version < std::make_tuple(3, 11)) {
+        // in GR 3.11, rename K->k, INTER->interleaver_taps and
+        // DEINTER->deinterleaver_taps
+        interleaver_bind.def("K", &interleaver::k, D(interleaver, k))
+            .def("INTER",
+                 &interleaver::interleaver_indices,
+                 D(interleaver, interleaver_indices))
+            .def("DEINTER",
+                 &interleaver::deinterleaver_indices,
+                 D(interleaver, deinterleaver_indices));
+        interleaver_bind.def(
+            py::init<unsigned int, std::vector<int, std::allocator<int>> const&>(),
+            py::arg(length_arg),
+            py::arg(interleaver_arg),
+            D(interleaver, interleaver, 2));
```

Comment:
this fails on `-Werror` builds because the deprecation warning gets emitted whether or not we actually bind or not. Need some precision preprocessor programming…